### ISSUE TITLE
bug(youtube): swipe brightness control on HDR videos are acting funky

### ISSUE THREAD

**Issue by SodaWithoutSparkles** [ReVanced/revanced-patches#748]
### Type

Error at runtime

### Bug description

Swipe brightness is acting funky on HDR videos, possibly interfering with HDR auto brightness

### Steps to reproduce

1. Patch YT 17.36.37 with manager 0.0.30 with all patches except debugging, auto repeat, wide search bar
2. Enable swipt brightness and sound control
3. Play a sample HDR video like [this one](https://youtu.be/gLcpjY3Gm2E) or [this one](https://youtu.be/mkggXE5e2yk) or [this one](https://youtu.be/jECIZpEsX5s)
4. Fullscreen the video and swipe to control brightness on the left side
5. Brightness control is not working normally

### Relevant log output

```shell
N/A
```

### Screenshots or videos

video resolution reduced for uploading, also muted

https://user-images.githubusercontent.com/64138578/194756447-2b9da99e-583e-4f5c-9d1c-ae5889fb00b2.mp4

### Solution

Swipe brightness ctrl should be prioritized before HDR auto brightness. 

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by its0x08** [comment]
Works perfectly on MIUI 13.0.4 POCO F3.

**Comment by SodaWithoutSparkles** [comment]
Strange, I am also on MIUI 13.0.5 Amdroid 12 POCO X4 GT

**Comment by its0x08** [comment]
I just confirmed that the video you mentioned makes YouTube works weird!

I answered above based on other hdr videos 🤔

**Comment by its0x08** [comment]
> Strange, I am also on MIUI 13.0.5 Amdroid 12 POCO X4 GT
> 
> @its0x08 did you enable HDR auto brightness under settings > revanced > miscellaneous ?

Yes I did. The thing is that I tested it using another video mot the one mentioned on the issue.

Now I see that the same problem exists on POCO F3 MIUI 13.0.4 Android 12
Latest Revanced toolkit ..

**Comment by SodaWithoutSparkles** [comment]
@its0x08 I have edited the top comment for 2 more videos that I have experienced the same issue. could you check those as well? the last one use VP9 codec if that matters, but I still get the funky swipe ctrls, even if my hands are way off the light sensor.

Could you send the vid that u used 4 testing?

**Comment by its0x08** [comment]
I confirm that with all three YouTube behaves the same.

**Comment by jonas-hurst** [comment]
As requested in https://github.com/revanced/revanced-patches/issues/745#issuecomment-1272568414, I reproduced this on LG V30 running Android 9. I can only reproduce it to a certain extent:

1. Right after opening video, I can change brightness by swiping without a problem
2. I tap the video settings after changing brightness (little gear top right corner in video player), then the brightness is reset to system setting
3. I tap video settings and change video quality, then things get weird: Brightness changes accordingly to my swipe on the screen, but as soon as I release my finger from the screen, brightness is reset to system setting.

I hope I was able to help you out.

**Comment by SodaWithoutSparkles** [comment]
New Findings: This behaviour exists no matter "Enable HDR auto brightness" under settings > misc is enabled or disabled. I have cleared the app from recent tasks before repeating the test.

I did not need to click the gear icon to have this funky brightness swipe

**Comment by SodaWithoutSparkles** [comment]
This is not fixed, as of:

- patch 2.97.0
- YT 17.41.37
- manager 0.0.36

**Comment by gimmee94** [comment]
This bug is still persistent and so annoying LOL.

**Comment by oSumAtrIX** [comment]
HDR automatically adjusts brightness, not sure if this a bug

**Comment by SodaWithoutSparkles** [comment]
> HDR automatically adjusts brightness, not sure if this a bug

It should only do so when swipe brightness control was set to auto.

**Comment by oSumAtrIX** [comment]
Isn't HDR controlled by the device?

**Comment by SodaWithoutSparkles** [comment]
> Isn't HDR controlled by the device?

I dont really know. Why does that even matter anyway?

It is counter-intuitive that the swipe-brightness-control overrides system brightness in all videos except HDR ones. Swipe brightness controls serves as a override to the brightness, and so should HDR content too. 

Even if that is not achievable, the scroll action should provide feedback to the user by locking the controls at auto, instead of seemingly scrolling but snapping back causing confusion.

**Comment by 54m4d** [comment]
Why was this issue closed without any clarification? Has it been resolved?

**Comment by oSumAtrIX** [comment]
@SodaWithoutSparkles HDR overrides brightness. If you have HDR enabled, brightness controls won't work because HDR should be controlled by the device which is why it matters.

**Comment by SodaWithoutSparkles** [comment]
> @SodaWithoutSparkles HDR overrides brightness. If you have HDR enabled, brightness controls won't work because HDR should be controlled by the device which is why it matters.

But it does work, at least on my device. Maybe I should get another device to physically record it.

**Comment by oSumAtrIX** [comment]
What does "it works" mean? HDR is supposed to change the brightness:

![image](https://github.com/ReVanced/revanced-patches/assets/13122796/9061130e-4130-4485-8951-eee788ab774a)

You are not supposed to override it.

**Comment by SodaWithoutSparkles** [comment]
Thats the issue. Some people would like to watch videos that are only available in HDR in their beds w/o blinding themselves.

**Comment by oSumAtrIX** [comment]
Swipe gestures is not supposed to override HDR settings in any way. For this, another patch can be made.

Can you change the brightness from your phone when watching HDR?

**Comment by SodaWithoutSparkles** [comment]
> Can you change the brightness from your phone when watching HDR?

Yes, both from the quick settings menu and with quick swipes in the video player.

**Comment by 54m4d** [comment]
And what is the purpose of this option? it doesn't make any difference if it's toggled or not.

![Screenshot_20230722_225246_YouTube](https://github.com/ReVanced/revanced-patches/assets/52612348/3662f913-cac3-4535-b2c7-0168a08af58a)

**Comment by oSumAtrIX** [comment]
Then the patch is broken, open an issue

**Comment by 54m4d** [comment]
Could you quickly verify it before i open the issue? just to make sure it's not working.

**Comment by 54m4d** [comment]
@oSumAtrIX I see that it works fine with other's patches, i see no reason why not to approach this in the same way they did?

https://github.com/ReVanced/revanced-patches/assets/52612348/99fe2ac4-87da-448e-8ba5-db2906da26a2
